src/OrchardCore.Modules/OrchardCore.AdminMenu/AdminNodes/LinkAdminNodeDriver.cs

```diff
@@ -23,20 +41,73 @@ public override IDisplayResult Edit(LinkAdminNode treeNode)
                 model.LinkText = treeNode.LinkText;
                 model.LinkUrl = treeNode.LinkUrl;
                 model.IconClass = treeNode.IconClass;
+                model.SelectedItems = new List<VueMultiselectItemViewModel>();
+
+                var nameList = new List<string>();
+                foreach (var permission in treeNode.Permissions)
+                {
+                    nameList.Add(permission.Name); 
+
+                    model.SelectedItems.Add(new VueMultiselectItemViewModel
+                    {
+                        Id = permission.Name,
+                        DisplayText = $"{permission.Name} - {permission.Description}"
+                    });
+                }
+                model.PermissionIds = string.Join(",", nameList);
+
             }).Location("Content");
         }
 
         public override async Task<IDisplayResult> UpdateAsync(LinkAdminNode treeNode, IUpdateModel updater)
         {
             var model = new LinkAdminNodeViewModel();
-            if (await updater.TryUpdateModelAsync(model, Prefix, x => x.LinkUrl, x => x.LinkText, x => x.IconClass))
+            if (await updater.TryUpdateModelAsync(model, Prefix, x => x.LinkUrl, x => x.LinkText, x => x.IconClass, x => x.PermissionIds))
             {
                 treeNode.LinkText = model.LinkText;
                 treeNode.LinkUrl = model.LinkUrl;
                 treeNode.IconClass = model.IconClass;
+
+                var modifiedPermissions= (model.PermissionIds == null? new string[0] : model.PermissionIds.Split(',', StringSplitOptions.RemoveEmptyEntries));
+                //clear the old permissions to insert all every time
+                treeNode.Permissions.Clear();
+                //change permissions only if one is inserted
+                if(modifiedPermissions.Length > 0)
+                {
+                    var permissions = await GetInstalledPermissionsAsync();
+
+                    foreach (var permissionName in modifiedPermissions)
+                    {
+                        var perm = permissions.Where(p => p.Name == permissionName).FirstOrDefault();
+                        
+                        if(perm != null)
+                        {
+                            treeNode.Permissions.Add(perm);
+                        }
+                    }
+                }
             };
 
             return Edit(treeNode);
         }
+
+        private async Task<IEnumerable<Permission>> GetInstalledPermissionsAsync()
```

Comment:
This could be moved in an abstraction module.

Reply:
`OrchardCore.Infrastructure.Abstractions`? I create a service to do this Job and load the service where it need? About `vue-multiselect-wrapper.js`, what did you think? I have copied it from the `OrchardCore.ContentFields` module but now I believe exist 2 or 3 copy of this file in the project.

Reply:
Yes that would make sense PermissionsService.cs